build/utils/yarn_test.go

```diff
@@ -24,24 +25,76 @@ func TestGetYarnDependencyKeyFromLocator(t *testing.T) {
 	}
 }
 
-func TestGetYarnDependencies(t *testing.T) {
+func TestGetYarnDependenciesV2(t *testing.T) {
+	// This test creates and tests a yarn project with yarn version of 3.1.1
+	CheckGetYarnDependencies(t, "v2", []string{"json@npm:9.0.6", "react@npm:18.2.0", "xml@npm:1.0.1"})
+}
+
+func TestBuildYarnDependenciesV1(t *testing.T) {
+	// This test creates and tests a yarn project with yarn version of 1.22.19
+	CheckGetYarnDependencies(t, "v1", []string{"json@9.0.6", "react@18.2.0", "xml@1.0.1"})
+}
+
+func CheckGetYarnDependencies(t *testing.T, versionDir string, expectedLocators []string) {
 	// Copy the project directory to a temporary directory
 	tempDirPath, err := utils.CreateTempDir()
 	assert.NoError(t, err, "Couldn't create temp dir")
 	defer func() {
 		assert.NoError(t, utils.RemoveTempDir(tempDirPath), "Couldn't remove temp dir")
 	}()
-	testDataSource := filepath.Join("..", "testdata", "yarn")
+	testDataSource := filepath.Join("..", "testdata", "yarn", versionDir)
 	testDataTarget := filepath.Join(tempDirPath, "yarn")
 	assert.NoError(t, utils.CopyDir(testDataSource, testDataTarget, true, nil))
 
+	// collecting and creating arguments for command
 	executablePath, err := GetYarnExecutable()
 	assert.NoError(t, err)
 	projectSrcPath := filepath.Join(testDataTarget, "project")
-	dependenciesMap, root, err := GetYarnDependencies(executablePath, projectSrcPath, &PackageInfo{Name: "build-info-go-tests", Version: "v1.0.0"}, &utils.NullLog{})
+	pacInfo := PackageInfo{Name: "build-info-go-tests", Version: "v1.0.0", Dependencies: make(map[string]string), DevDependencies: make(map[string]string)}
+	pacInfo.Dependencies["react"] = "18.2.0"
+	pacInfo.Dependencies["xml"] = "1.0.1"
+	pacInfo.DevDependencies["json"] = "9.0.6"
+	dependenciesMap, root, err := GetYarnDependencies(executablePath, projectSrcPath, &pacInfo, &utils.NullLog{})
+
+	// general checks
 	assert.NoError(t, err)
 	assert.NotNil(t, root)
-	assert.True(t, strings.HasPrefix(root.Value, "build-info-go-tests@"))
+
+	// checking root
+	assert.True(t, strings.HasPrefix(root.Value, "build-info-go-tests"))
 	assert.Equal(t, "v1.0.0", root.Details.Version)
-	assert.Len(t, dependenciesMap, 3)
+	for _, dependency := range root.Details.Dependencies {
+		assert.Contains(t, expectedLocators, dependency.Locator)
+	}
+
+	// checking dependencyMap
+	// NOTICE: the test uses fixed package versions. if for some reason dependencies in those packages (with those versions) changes
+	// (i.e. new dependencies added) make sure to fix the checks below
+	assert.Len(t, dependenciesMap, 6)
+	for key, val := range dependenciesMap {
+		switch {
```

Comment:
you can split the key to get the prefixed string and then run a simple` switch key { ... }`